core/src/main/java/org/mskcc/cbio/portal/scripts/ImportGeneData.java

```diff
@@ -607,96 +683,105 @@ static void importSuppGeneData(File suppGeneFile, String referenceGenomeBuild) t
+
+            // using a real options parser, helps avoid bugs
+            OptionParser parser = new OptionParser();
+            OptionSpec<Void> help = parser.accepts( "help", "print this help info" );
+            parser.accepts( "genes", "ncbi genes file" ).withRequiredArg().describedAs( "ncbi_genes.txt" ).ofType( String.class );
+            parser.accepts( "hgnc", "hgnc genes file" ).withRequiredArg().describedAs( "hgnc_genes.txt" ).ofType( String.class );
+            parser.accepts( "supp-genes", "alternative genes file" ).withRequiredArg().describedAs( "supp-genes.txt" ).ofType( String.class );
+            parser.accepts( "microrna", "microrna file" ).withRequiredArg().describedAs( "microrna.txt" ).ofType( String.class );
+            parser.accepts( "gtf", "gtf file for calculating and storing gene lengths" ).withRequiredArg().describedAs( "gencode.<version>.annotation.gtf" ).ofType( String.class );
+            parser.accepts( "genome-build", "genome build eg GRCh38" ).withRequiredArg().describedAs( "genome build" ).ofType( String.class );
+            parser.accepts( "species", "different kinds of organisms eg. humna").withRequiredArg().describedAs( "species" ).ofType( String.class );
+            String progName = "importGenes";
+            OptionSet options = null;
+            try {
+                options = parser.parse( args );
+            } catch (OptionException e) {
+                throw new UsageException(progName, description, parser,
+                    e.getMessage());
+            }
+
+            if( options.has( help ) ){
+                throw new UsageException(progName, description, parser);
+            }
+
+            ProgressMonitor.setConsoleMode(true);
+
+            if(options.has("genes")) {
+                File geneFile = new File((String) options.valueOf("genes"));
+                System.out.println("Reading gene data from:  " + geneFile.getAbsolutePath());
+                int numLines = FileUtil.getNumLines(geneFile);
+                System.out.println(" --> total number of lines:  " + numLines);
+                ProgressMonitor.setMaxValue(numLines);
+                MySQLbulkLoader.bulkLoadOn();
+                ImportGeneData.importData(geneFile, (String)options.valueOf("genome-build"));
+                MySQLbulkLoader.flushAll(); //Gene and gene_alias should be updated before calculating gene length (gtf)!
+            }
+
+            if(options.has("hgnc")) {
+                File geneFile = new File((String) options.valueOf("hgnc"));
+                System.out.println("Reading gene data from:  " + geneFile.getAbsolutePath());
+                int numLines = FileUtil.getNumLines(geneFile);
+                System.out.println(" --> total number of lines:  " + numLines);
+                ProgressMonitor.setMaxValue(numLines);
+                MySQLbulkLoader.bulkLoadOn();
+                ImportGeneData.importHGNCData(geneFile, (String)options.valueOf("genome-build"));
```

Comment:
another possible improvement : validation of the genome-build string. For now we have some expectations of the format of chromosome identifiers, so we can handle "human-like" chromosome names. hg, mm, rn. But this breaks down for some other species (drosophila has "3L" and "3R" for instance, or Caenorhabditis elegans uses roman numerals).